pkg/scheduler/frameworkext/helper/synced_eventhandler.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helper
+
+import (
+	"reflect"
+
+	"k8s.io/client-go/tools/cache"
+)
+
+type forceSyncEventHandler struct {
+	handler cache.ResourceEventHandler
+	syncCh  chan struct{}
+}
+
+func newForceSyncEventHandler(handler cache.ResourceEventHandler) *forceSyncEventHandler {
+	return &forceSyncEventHandler{
+		handler: handler,
+		syncCh:  make(chan struct{}, 1),
+	}
+}
+
+func (h *forceSyncEventHandler) syncDone() {
+	close(h.syncCh)
+}
+
+func (h *forceSyncEventHandler) waitForSyncDone() {
+	<-h.syncCh
+}
+
+func (h *forceSyncEventHandler) OnAdd(obj interface{}) {
+	h.waitForSyncDone()
```

Comment:
OnAdd may be called repeatedly? this require eventHandler idempotent. We need add comment and check every eventhandler or find alternative approach.

Reply:
Yes, OnAdd will be called repeatedly with same object.  There is currently no particularly good way to avoid onAdd being called repeatedly. A feasible method is to record the resourceVersion of each object. Whenever onAdd comes in, check whether the version is greater than the recorded resourceVersion. If it is greater, the processing is allowed to continue.